### PR TITLE
fetch: remove use of `args` from CLI parser

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

See discussion at #15382.

### CODE REVIEW

**Review by Bo98 — APPROVED** [review]

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Regarding https://github.com/Homebrew/brew/pull/15382#discussion_r1188024084.

I have also said this in the past, since we are using `args` somewhere way deep in `GitHub` which get's passed using `**options` and is very hard to follow.

File: Library/Homebrew/fetch.rb [hunk 1]
```diff
@@ -4,16 +4,23 @@
...
+        build_from_source_formulae: T::Array[String],
+      ).returns(T::Boolean)
+    }
+    def fetch_bottle?(formula, force_bottle:, bottle_tag:, build_from_source_formulae:)
       bottle = formula.bottle
 
-      return true if args.force_bottle? && bottle.present?
-      return true if args.bottle_tag.present? && formula.bottled?(args.bottle_tag&.to_sym)
+      return true if force_bottle && bottle.present?
+      return true if bottle_tag.present? && formula.bottled?(bottle_tag)
```
Comment (reitermarkus) [inline Library/Homebrew/fetch.rb:19]:
```suggestion
      return true if bottle_tag && formula.bottled?(bottle_tag)
```

File: Library/Homebrew/fetch.rb [hunk 2]
```diff
@@ -4,16 +4,23 @@
...
+        bottle_tag:                 T.nilable(Symbol),
+        build_from_source_formulae: T::Array[String],
+      ).returns(T::Boolean)
+    }
+    def fetch_bottle?(formula, force_bottle:, bottle_tag:, build_from_source_formulae:)
       bottle = formula.bottle
 
-      return true if args.force_bottle? && bottle.present?
-      return true if args.bottle_tag.present? && formula.bottled?(args.bottle_tag&.to_sym)
+      return true if force_bottle && bottle.present?
```
Comment (reitermarkus) [inline Library/Homebrew/fetch.rb:18]:
```suggestion
      return true if force_bottle && bottle
```

File: Library/Homebrew/fetch.rb (hunk 1 above)
Comment (carlocab) [reply Library/Homebrew/fetch.rb:19]: `bottle_tag` can be an empty string so dropping `#present?` looks like it could change the behaviour of this method, which I'd rather not do.
